### PR TITLE
Adding a script to check for line endings (valid UNIX ending)

### PR DESCRIPTION
**Pull request by kushalsingh007**
-Created a bash script using grep to check for carriage return at line endings.
-Fix for issue #2281.

### CODE REVIEW

**Comment by kushalsingh007** [comment]
Yet again the build fail is due to Travis being flaky , kindly re-run the build again :)

**Comment by authmillenon** [comment]
Actually it fails because there are trailing whitespaces in your script. Please remove them ;)

**Comment by authmillenon** [comment]
(It's flaky when some random board fails. The static tests are quite stable)

Speaking of: could you add your script to them (add a call to your script to `dist/tools/travis-scripts/build_and_test.sh`). Also, I think it would be best if it only applies to new and changed files. Check out the other check scripts as for example `dist/tools/whitespacecheck/check.sh` or `dist/tools/cppcheck/check.sh` for how to do this.

**Comment by kushalsingh007** [comment]
Ok , I will have a look at it and make sure that the check applies only to new files :)

**Comment by kushalsingh007** [comment]
@authmillenon : I have added a new script which checks only for the newly added files and added a call to it in build_and_run.sh. However  I do not have the permission to modify build_and_run.sh ( and that is why the static case fails ) . So have a look at it .

File: dist/tools/travis-scripts/build_and_test.sh [hunk 1]
```diff
@@ -45,6 +45,15 @@ then
         ./dist/tools/externc/check.sh master
         RESULT=$(set_result $? $RESULT)
 
+       ./dist/tools/endingcheck/check.sh master --diff-filter=MR
+       RESULT=$(set_result $? $RESULT)
+
+       ./dist/tools/endingcheck/check.sh master --diff-filter=AC
+       RESULT=$(set_result $? $RESULT)
+
+       ./dist/tools/endingcheck/check.sh riot/master
```
Comment (authmillenon) [inline dist/tools/travis-scripts/build_and_test.sh]: One call of your script with `riot/master` (the current master at RIOT-OS/RIOT) and `--diff-filter=MA` (only look at modified [M] and added [A] files) should suffice (source: http://git-scm.com/docs/git-diff).

**Comment by authmillenon** [comment]
> However I do not have the permission to modify build_and_run.sh ( and that is why the static case fails ). 

What do you mean? You obviously modified it (see my comment above). So how do you come to the impression you don't have any permission to do so?  :confused:

**Comment by kushalsingh007** [comment]
That's because it is the reason build failed

**Comment by authmillenon** [comment]
Ah... Your script is not executable. That's why Travis is saying the permissions were denied (because it has no permission to execute it). Git has an understanding of UNIX permission flags so you have to give your script execution permission:

```bash
chmod +x ./dist/tools/endingcheck/check.sh
```

**Comment by authmillenon** [comment]
(and than of course add this change to git ;) 

```bash
git add ./dist/tools/endingcheck/check.sh
git commit
```

**Comment by kushalsingh007** [comment]
Done , the changes and the build

File: dist/tools/endingcheck/check.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#
+# Copyright (C) 2015 Kushal Singh <kushal.spiderman.singh@gmail.com>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+BRANCH=${1}
+FILEREGEX='\.([sScHh]|cpp)$'
```
Comment (authmillenon) [inline dist/tools/endingcheck/check.sh]: I would actually include all files, when I think about it, not just source code.

**Comment by authmillenon** [comment]
Please squash only when the review is done, otherwise it can get a little bit confusing with bigger changes ;-)

**Comment by kushalsingh007** [comment]
Ok , I'll keep that in mind from next time. So , can I squash now  to change the file regex ?

File: dist/tools/endingcheck/check.sh (hunk 2 above)
Comment (kushalsingh007) [inline dist/tools/endingcheck/check.sh]: There are things like images which should be excluded from it.

**Comment by authmillenon** [comment]
No, just mark the commits as squashable (I usually use a `[SQUASH ME]` tag in front of the commit description) and squash it, if you get the ACK ;-)

**Comment by authmillenon** [comment]
> (I usually use a [SQUASH ME] tag in front of the commit description)

See #2546 for example

File: dist/tools/travis-scripts/build_and_test.sh [hunk 3]
```diff
@@ -45,6 +45,11 @@ then
         ./dist/tools/externc/check.sh master
         RESULT=$(set_result $? $RESULT)
 
+       RESULT=$(set_result $? $RESULT)
```
Comment (gebart) [inline dist/tools/travis-scripts/build_and_test.sh]: Why is this duplicate line here?

File: dist/tools/travis-scripts/build_and_test.sh [hunk 4]
```diff
@@ -45,6 +45,11 @@ then
...
+
+       ./dist/tools/endingcheck/check.sh riot/master --diff-filter=MA
+       RESULT=$(set_result $? $RESULT)
```
Comment (gebart) [inline dist/tools/travis-scripts/build_and_test.sh]: Increase indentation by 1 space

**Comment by kushalsingh007** [comment]
@OlegHahm : Done squashing.

**Pull request merged**